.github/workflows/wheels.yml

```
@@ -77,11 +79,16 @@ jobs:
         # should also be able to do multi-archs on a single entry, e.g.
         # [windows-2019, win*, "AMD64 x86"]. However, those two require a different compiler setup
         # so easier to separate out here.
-        - [ubuntu-22.04, manylinux, x86_64]
-        - [ubuntu-22.04, musllinux, x86_64]
-        - [macos-11, macosx, x86_64]
-        - [macos-14, macosx, arm64]
-        - [windows-2019, win, AMD64]
+        - [ubuntu-22.04, manylinux, x86_64, "", ""]
+        - [ubuntu-22.04, musllinux, x86_64, "", ""]
+        - [macos-13, macosx, x86_64, openblas, "10.9"]
+        # Build x86 Accelerate wheel targeting macOS 13 just for testing
+        - [macos-13, macosx, x86_64, accelerate, "13.0"]
+        # Build x86 Accelerate wheel targeting macOS 14 to release
+        - [macos-13, macosx, x86_64, accelerate, "14.0"]
```

Can you try build and test of macos_x86_64 on the macos-14 image?

```suggestion
        - [macos-14, macosx, x86_64, accelerate, "14.0"]
```

---

I considered this but concluded that the test results under x86 emulation are also meaningless. For instance, I used `arch -x86_64` on an M1 runner to [test an x86 OpenBLAS wheel built on an x86 runner](https://github.com/thalassemia/test-scipy-wheels/actions/runs/8741694778/job/23988284586). This resulted in 7 test failures even though the wheel passed all tests on the x86 runner. I saw the exact same tests fail when [testing the x86 Accelerate wheel](https://github.com/thalassemia/test-scipy-wheels/actions/runs/8730701880/job/23954942627), so we're likely catching emulation bugs instead of actual bugs.

I did manage to figure out a better compromise. It turns out the only thing preventing the wheel from installing on macOS 13 was the `14_0` in the file name. By changing that to `13_0`, I got the wheel to install and run the test suite on a macOS 13 runner. I then added a new step to rename the wheel back to target macOS >= 14. The [resulting test failures](https://github.com/thalassemia/scipy/actions/runs/8743574037/job/23994426272) are the same as those from the now-removed build job targeting macOS 13.

---

FWIW it's fairly straightforward to set up a conda environment that uses emulated python-x86_64 on an M1 machine. See https://stackoverflow.com/questions/71691598/how-to-run-python-as-x86-with-rosetta2-on-arm-macos-machine.

---

Ah that's a good point. I was calling the native ARM pip and python to install x86 wheels with my `arch -x86_64` test, deviating even further from a true x86 environment. I could repeat the test in a Conda environment as you suggested, but I would still be worried that the emulation either causes new bugs or masks bugs that exist on real x86_64 hardware (especially since Accelerate is a proprietary system library whose x86_64 version cannot be pulled from conda for testing).

---

I used this approach to test a macos-x86_64-accelerate-14.0 wheel that you built on your fork. I saw 8 test failures :-(.
I then tried to build from that kind of conda environment but compilation doesn't get very far because of issues with the toolchain.

---

I also saw toolchain issues in my early attempts at cross-compiling (didn't know about the Conda trick). I could not figure out how to get an ARM-to-x86 gfortran cross compiler for macOS and ended up scrapping the whole idea.